[RFC] Run old tests in :terminal

ZyX-I opened the pull request:
Reasoning:

- No need to check for terminal size.
- No need to mess with terminal title.
- Allows old tests to timeout with the rest of the build proceeding.
- Less and less messy output to travis log.
- Opens a path allowing old tests run in parallel. Though last bit needs test
  refactoring.

ZyX-I committed:
oldtests: Run old tests in :terminal

Reasoning:

- No need to check for terminal size.
- No need to mess with terminal title.
- Allows old tests to timeout with the rest of the build proceeding.
- Less and less messy output to travis log.
- Opens a path allowing old tests run in parallel. Though last bit needs test
  refactoring.

justinmk commented on src/nvim/testdir/runnvim.sh:
```diff
@@ -0,0 +1,74 @@
+  # src   testdir
+  #    nvim
+  . ../../../ci/common/suite.sh
+  . ../../../ci/common/test.sh
```
can `$CI_DIR` be used here?

ZyX-I commented:
The thing works in travis, the only failure is unrelated as I saw that test flacky. (Does not work locally though.) AV still has the thing queued. QB has only timer failures and functional tests timeout, and it does not look like it runs anything I touched at all.

Though I think I will modify Makefile to produce exactly one line of output per old test and nothing more. And probably disable `set -x`, as 10 000 lines limit is still being hit.

ZyX-I committed:
oldtests: Make test output least verbose

Also removes `-` before `rm -rf` as `-f` is supposed to already suppress error.

ZyX-I commented:
(That local failure with test_cmdline is basically the same I experienced earlier: that test for some reason searches fileXname in /usr/include and some package has set up recursive symbolic link there so search never actually ends.)

justinmk commented:
> QB has only timer failures and functional tests timeout, and it does not look like it runs anything I touched at all.

Yes, QB doesn't run anything in `src/nvim/testdir`.

ZyX-I commented:
CI almost succeeded, is under 10 000 lines (though without much space left) and it does not look like any of the failures are related.

justinmk commented:
@ZyX-I This is awesome. I wonder if we should add `oldtest` to the `test` target now?

ZyX-I commented:
The parent has its `$VIMRUNTIME` set for the child, it does not use it itself. I do not know why AV is printing that, these patterns are `find` argument which is not supposed to print anything upon failure. And everything is properly quoted as far as I see.

janlazo commented:
@ZyX-I Windows has its own incompatible `find.exe`. Prepending in `C:\msys64\usr\bin` to PATH instead of appending should fix it but `/usr/bin/find` should work as well.
